app/lib/katello/logging.rb

```diff
@@ -9,6 +9,33 @@ def self.time(message, data: {}, logger: Rails.logger, level: :info)
       data_string = data.map { |k, v| "#{k}=#{v}" }.join(' ')
 
       logger.send(level, "#{message} #{data_string}")
+      "#{message} #{data_string}"
+    end
+
+    class Timer
+      def initialize(key = "default")
+        @key = key
+        @start_time = Time.now
+        @@timers ||= {}
+        @@timers[key] = self
+      end
+
+      def start
+        Rails.logger.info "Timer #{@key} starting at #{Time.now}"
+        @start_time = Time.now
+        self
+      end
+
+      def stop
+        duration = (Time.now - @start_time).truncate(2)
+        @start_time = nil
+        Rails.logger.info "Timer #{@key} stopping at #{Time.now}: #{duration} sec"
+        "Timer #{@key} stopping at #{Time.now}: #{duration} sec"
+      end
+
+      def self.find_by_key(key)
+        @@timers[key]
```

Comment:
since this depends on the local process that is running, imagine setting some timer in the plan phase of a dynflow action and then stopping it in the finalize (this wouldn't work at all), but it might be better to handle that gracefully?  I.e. Return a new timer instead of nil if one isn't found with a given key (maybe with a warning in that timer's message that the data may not be accurate)?